scripts/check-repository-health.js

```diff
@@ -0,0 +1,246 @@
+      return repos[path].component;
+    }
+  }
+  throw new Error(`Repository with name "${name}" not found`);
+}
+
+async function getAptPackages(repo, arch, errors, proposedFixes) {
+  const url = `${repo.url}/dists/${repo.distribution}/${repo.component}/binary-${arch}/Packages.gz`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch ${url}: ${response.status} ${response.statusText}`,
+    );
+  }
+  const data = await gunzipAsync(await response.arrayBuffer());
+  const aptPackages = new Map();
+  let pkgName = undefined;
+  let pkgVersion = undefined;
+  data
+    .toString()
+    .split("\n")
+    .forEach((line) => {
+      // console.debug(line);
+      if (line.startsWith("Package: ")) {
+        pkgName = line.substring("Package: ".length);
+      } else if (line.startsWith("Version: ")) {
+        pkgVersion = line.substring("Version: ".length);
+      } else if (line == "") {
+        if (pkgName && pkgVersion) {
+          if (aptPackages.has(pkgName)) {
+            errors.push(
+              `Duplicate package: "${pkgName}" when parsing Packages file for "${repo.name}" for "${arch}"`,
+            );
+            proposedFixes.push(
+              `# Duplicate package "${pkgName}" will likely be removed automatically once the cron job responsible for cleaning older versions of packages kicks in on the aptly server.`,
+            );
+          } else {
+            // Only add the package version.
+            // This also has a desired side-effect of only keeping the latest version since aptly seems to sort the Packages file
+            aptPackages.set(pkgName, {
+              version: pkgVersion,
+              repo: repo.name,
+            });
+          }
+        }
+        pkgName = undefined;
+        pkgVersion = undefined;
+      }
+    });
+  if (pkgName || pkgVersion) {
+    console.error(`Incomplete package entry in ${url}`);
+    process.exit(1);
+  }
+
+  return aptPackages;
+}
+
+async function getTermuxPackages(arch, errors, proposedFixes) {
+  const termuxPackages = new Map();
+  const data = await readFile(
+    `./scripts/apt-packages-list-${arch}.txt`,
+    "utf8",
+  );
+  data
+    .trim()
+    .split("\n")
+    .forEach((line) => {
+      let [pkgName, pkgRepo, pkgVersion, pkgMayHaveStaticSubpkg] =
+        line.split(" ");
+      if (termuxPackages.has(pkgName)) {
+        errors.push(`Duplicate package in termux-packages: "${pkgName}"`);
+        proposedFixes.push(
+          `# Duplicate package "${pkgName}" earlier found in "${termuxPackages.get(pkgName).repo}" also in "${pkgRepo}" needs to be removed from termux-packages`,
+        );
+      }
+      termuxPackages.set(pkgName, {
+        version: pkgVersion,
+        repo: pkgRepo,
+        mayHaveStaticSubpkg: pkgMayHaveStaticSubpkg === "true",
+      });
+    });
+  return termuxPackages;
+}
+
+async function getErrorsForArch(arch) {
+  const errors = [];
+  const proposedFixes = [];
```

Comment:
Won't block on this, but consider splitting up proposed fixes into the ones that are essentially a shell script and the ones that are instructions for a human to go follow on github. It seems right that this script identifies both but unhelpful to actually provide the latter as comments in the former.